### DISCUSSION TITLE
embeded UI on faust2bela

### DISCUSSION THREAD

**Discussion by PascalFaivre** [grame-cncm/faust#831] [Q&A]
For my usage, i'm working on a project allowing to use I2C screens to display and edit parameters on bela platform :
https://github.com/PascalFaivre/MiniDisplay_UI
I would like to know if you see an interest in integrating this feature into Faust2bela.

**Comment by sletz** [comment]
Possibly yes: how do see the integration?

**Comment by PascalFaivre** [comment]
use the 'declare' directive for screen configuration like :  declare minidiplay_ui "{ 'Label of screen' : { 'type' : 'type of screen' ; 'width' : L ; 'height' : H ; 'uppin' : x ; 'downpin' : y ; 'leftpin' : z ; 'rightpin' : a ; 'selpin' : b } }"
use the group primitives for page declaration with metadata for screen identification like : [MINIDISPLAY_UI:Label of screen]
use the widgets for parameters declarations with metadata for configuration like : [MINIDISPLAY_UI : {'type':'type of display'; 'editable' : 1}].
the MiniDisplay_UI library must be added separalty or the scripts faust2bela add the library to the project.

**Comment by sletz** [comment]
I would say we are interested if the development can be done in a module way, and making this new feature easily configurable and deployable.

**Comment by PascalFaivre** [comment]
by 'a module way' do you mean : a wrapper with the construction's logic in a separate file and option in the faust2bela bash file for activate the feature and deploy the librarie ?

**Reply by sletz** [reply]
Yes this kind of thing.

**Comment by PascalFaivre** [comment]
Ok, I'm finishing support for Trill sensors and I'm studying that.
